### PR TITLE
show progress while running the analysis server

### PR DESCRIPTION
**Pull request by devoncarew**
- show progress while running the analysis server

@Hixie

### CODE REVIEW

**Comment by Hixie** [comment]
LGTM

File: packages/flutter_tools/lib/src/commands/analyze.dart [hunk 1]
```diff
@@ -299,18 +300,22 @@ class AnalyzeCommand extends FlutterCommand {
...
   int lastErrorCount = 0;
+  Status analysisStatus;
 
   void _handleAnalysisStatus(AnalysisServer server, bool isAnalyzing) {
     if (isAnalyzing) {
+      analysisStatus?.cancel();
+
       if (firstAnalysis) {
-        printStatus('Analyzing ${path.basename(Directory.current.path)}...');
+        analysisStatus = logger.startProgress('Analyzing ${path.basename(Directory.current.path)}...');
```
Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: btw, this doesn't do the right thing if you're using --flutter-repo

Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: oops, will follow up

Comment (devoncarew) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: ah, the name it prints is wrong. It's assuming that you're in the root of the flutter repo

Comment (Hixie) [inline packages/flutter_tools/lib/src/commands/analyze.dart]: Yeah, I fix this in #4528.

**Pull request merged**